mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresUidProvider.java

```diff
@@ -0,0 +1,93 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.postgres.mail;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.LongStream;
+
+import javax.inject.Inject;
+
+import org.apache.james.mailbox.MessageUid;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.Mailbox;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.postgres.mail.dao.PostgresMailboxLastUidDAO;
+import org.apache.james.mailbox.store.mail.UidProvider;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import reactor.core.publisher.Mono;
+
+public class PostgresUidProvider implements UidProvider {
```

Comment:
Enable RLS. MailboxSession can be a parameter of the providers and passed by other mappers when creating this object.

Reply:
where do we get `MailboxSession`?
The API is `public abstract UidProvider getUidProvider()`
NOT `public abstract UidProvider getUidProvider(MailboxSession session)`

Reply:
We can modify the base interface in order to do so?

Reply:
yes, we can

Reply:
If we do I don't think we need tables for uid and modseq anymore right? can put them back to mailbox table?